### PR TITLE
Add forms to blog app

### PR DESCRIPTION
**Pull request by Bellagirl-maker**
- Create a method called current_user in your ApplicationController to make current user data available to all controllers.
- It will return the first user from the database.
- Don't worry about proper authentication at this time. You will add it in the upcoming projects.
- Create forms to perform the following functions:
- Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
- Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).
- Allow Users to add likes to Posts.

### CODE REVIEW

File: app/views/posts/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<h1>Create a new Post</h1>
+
+<%= form_with(model: @post, url: posts_path) do |form| %>
```
Comment (DammyShittu) [inline app/views/posts/new.html.erb]: - [ ] When I visit the new post route to create a post I get this error message. Kindly fix that. To check other functionalities I had to create a post from the rails console, so kindly fix this.

  image

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    @post = Post.find(params[:post_id])
+    @comment = @post.comment_entries.build(comment_params)
+    @comment.user = current_user
+
+    if @comment.save
+      redirect_to post_path(@post), notice: 'Comment was successfully created.'
+    else
+      render :new
+    end
+  end
```
Comment (DammyShittu) [inline app/controllers/comments_controller.rb]: - [ ] I cannot create a comment either. Kindly fix that. I think this has to do with the `comment_params` or so.

  image

File: app/views/shared/_like.html.erb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<%= form_with model: @like, url: "/users/#{params[:user_id]}/posts/#{params[:id]}/likes/create" do |form| %>
+  <%= form.submit "Like"%>
+<% end %>
```
Comment (DammyShittu) [inline app/views/shared/_like.html.erb]: - [ ] I cannot create a like too. Kindly fix that. You may need to ensure your route is well set up.

  image

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @Bellagirl-maker,

## Highlights
✔️ Current user in app controller

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (Bellagirl-maker) [reply app/controllers/comments_controller.rb]: @DammyShittu done with this change

File: app/views/shared/_like.html.erb (hunk 3 above)
Comment (Bellagirl-maker) [reply app/views/shared/_like.html.erb]: @DammyShittu done with this change

File: app/views/posts/new.html.erb (hunk 1 above)
Comment (Bellagirl-maker) [reply app/views/posts/new.html.erb]: @DammyShittu done with this change

File: app/controllers/comments_controller.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  def create
+    @post = Post.find(params[:post_id])
+    @comment = @post.comment_entries.build(comment_params)
+    @comment.user = current_user
+
+    if @comment.save
+      redirect_to post_path(@post), notice: 'Comment was successfully created.'
+    else
+      render :new
+    end
```
Comment (DuaneDave) [inline app/controllers/comments_controller.rb]: You have done a great deal addressing all previously found issues, however, when a comment is created, it doesn't show on the UI. The normal behaviour should after a comment is created, the user should be redirected to the post page so the comment can reflect there, but when I create a comment and manually navigate to the posting page the comment is not there. I recommend addressing this problem in order to satisfy the requirement.

I have recorded a video to show this behaviour:

[screen-capture.webm](https://github.com/Bellagirl-maker/blog_app/assets/84930355/4f90fc4c-b76e-4cef-a760-cdd5c8c0cfae)

**Review by DuaneDave — CHANGES_REQUESTED** [review]
Hi @Bellagirl-maker 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️ Likes can be created
✔️ Posts are created without any errors

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/comments_controller.rb (hunk 4 above)
Comment (Bellagirl-maker) [reply app/controllers/comments_controller.rb]: @DuaneDave Done with this change

**Review by DammyShittu — APPROVED** [review]
Hi @Bellagirl-maker,

## APPROVED 🟢

## Highlights
✔️ Project met all technical requirements

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**